src/Apps/MyCollection/Routes/EditArtwork/Components/MyCollectionArtworkFormImages.tsx

```diff
@@ -8,19 +9,50 @@ import {
   uploadMyCollectionPhoto,
 } from "Components/PhotoUpload/Utils/fileUtils"
 import { useFormikContext } from "formik"
-import { useEffect, useState } from "react"
+import { forwardRef, useEffect, useImperativeHandle, useState } from "react"
 import { FileRejection } from "react-dropzone"
 import { useSystemContext } from "System"
+import { LocalImage, storeArtworkLocalImages } from "Utils/localImagesHelpers"
 import { MyCollectionPhotoToPhoto } from "../Utils/artworkFormHelpers"
 import { ArtworkModel } from "../Utils/artworkModel"
 
-interface MyCollectionArtworkFormImagesProps {}
-
-export const MyCollectionArtworkFormImages: React.FC<MyCollectionArtworkFormImagesProps> = () => {
+export interface MyCollectionArtworkFormImagesComponentRef {
+  saveImagesToLocalStorage: (artworkId: string) => Promise<string | undefined>
+}
+export const MyCollectionArtworkFormImages = forwardRef<
+  MyCollectionArtworkFormImagesComponentRef
+>((_, formImagesRef) => {
   const [errors, setErrors] = useState<Array<FileRejection>>([])
+  const [localImages, setlocalImages] = useState<
+    Array<LocalImage & { photoID: string }>
+  >([])
   const { relayEnvironment } = useSystemContext()
   const { values, setFieldValue } = useFormikContext<ArtworkModel>()
 
+  const saveImagesToLocalStorage = async (artworkId: string) => {
+    try {
+      // Store the artwork's local images in local storage
+      //and remove unnecessary fields
+      return await storeArtworkLocalImages(
+        artworkId,
+        localImages.map(({ photoID, ...rest }) => rest),
+        IMAGES_LOCAL_STORE_KEY
```

Comment:
**question (non-blocking):** What do you think about removing the key as an argument? looking at the name of the function it's only purpose is to store local images anyway.

Reply:
of course